providers/certs.rb

```diff
@@ -85,14 +85,44 @@
                 else
                   raise "Error signing certificate #{new_resource.name}"
                 end
+
+                #pass the key and certificate to hopsmon
+                crt = {data: ::File.read("#{new_resource.path}/#{new_resource.name}.crt")}
+                kagent_param "/tmp" do
+                  executing_cookbook "kube-hops"
+                  executing_recipe "hopsmon"
+                  cookbook "hopsmonitor"
+                  recipe "prometheus"
+                  param "kube-crt"
+                  value "'" + JSON.generate(crt) + "'"
+                end
+
+                key = {data: ::File.read("#{new_resource.path}/#{new_resource.name}.key")}
+                kagent_param "/tmp" do
+                  executing_cookbook "kube-hops"
+                  executing_recipe "hopsmon"
+                  cookbook "hopsmonitor"
+                  recipe "prometheus"
+                  param "kube-key"
+                  value  "'" + JSON.generate(key) + "'"
+                end
+
+                ca = {data: ::File.read("#{node['certs']['dir']}/kube/certs/kube-ca.cert.pem")}
+                kagent_param "/tmp" do
+                  executing_cookbook "kube-hops"
+                  executing_recipe "hopsmon"
+                  cookbook "hopsmonitor"
+                  recipe "prometheus"
+                  param "kube-ca"
+                  value  "'" + JSON.generate(ca) + "'"
+                end
```

Comment:
This is not the correct place to put this. This provider is being used by all the K8s cervices that need a certificate, not just hopsmon. 
It probably works now because the hopsmon is the last certificate to be generated, but if someone adds a certificate after that, it's going to fail.

Reply:
I added an if statement